prefect_hightouch/api_client/api/__init__.py

```diff
@@ -0,0 +1,42 @@
+""" Contains methods for accessing the API """
+
+import functools
+from typing import Callable, TypeVar
+
+from typing_extensions import Concatenate, ParamSpec
+
+from prefect_hightouch.api_client.client import AuthenticatedClient
+from prefect_hightouch.credentials import HightouchCredentials
+
+C = ParamSpec("C")  # client function
+T = ParamSpec("T")  # task function
+R = TypeVar("R")  # The return type of the API function
+
+
+def _execute_endpoint(
```

Comment:
Not a big deal, but I think we can come up with a clearer name for this since this function doesn't execute the endpoint, the inner one does. I don't have a great suggestion, but maybe `@_convert_credentials_to_client`?

Reply:
I am a little hesitant to rename to just `@_convert_credentials_to_client` because it encompasses executing the endpoint.

If a user saw this with only `...` and a decorator that "converts credentials to client" it's a little a misleading I think.
![image](https://user-images.githubusercontent.com/15331990/200027068-48954dda-1018-414a-89c4-8c11f4da3e98.png)